tests/tests_twobody/test_orbit.py

```diff
@@ -1101,3 +1101,16 @@ def test_time_to_anomaly():
     tof = iss_180.time_to_anomaly(0 * u.deg)
 
     assert_quantity_allclose(tof, expected_tof)
+
+
+def test_issue_798():
+    epoch = Time("2019-11-10 12:00:00")
+    iss = Orbit.from_horizons(
+        "International Space Station", Sun, epoch=epoch, id_type="majorbody"
+    )
+    with pytest.raises(ValueError) as excinfo:
+        iss.change_attractor(Earth)
+    assert (
```

Comment:
I agree, we should rethink about the logic used here. https://github.com/poliastro/poliastro/blob/f4547d2b1b89eba41a9ee29b3fe35d9ba393c64c/src/poliastro/twobody/orbit.py#L502-L503

Reply:
The current logic covers this case:

https://github.com/poliastro/poliastro/blob/f4547d2b1b89eba41a9ee29b3fe35d9ba393c64c/tests/tests_twobody/test_orbit.py#L1061-L1067

Therefore, we can't just remove it, because it's legitimate. However, there should be a way to detect when an object is, in fact, orbiting a closer body.

Reply:
@iamabhishek0 My suggestion: let's create a test that asserts that the change *can* be made, mark it as `xfail`, and let's tackle that in a future pull request.